assets/components/src/plugin-settings/SettingsSection.js

```diff
@@ -91,45 +87,53 @@ const SettingsSection = props => {
 			props
 		);
 	};
+	let columns;
+	if ( fields.length % 3 === 0 ) {
+		columns = 3;
+	} else if ( fields.length % 2 === 0 ) {
+		columns = 2;
+	} else {
+		columns = 1;
+	}
```

Comment:
This calculation makes prime numbers (e.g. 5) always have 1 column. A better approach would be:

```suggestion
	let columns = 2;
	if ( fields.length % 3 === 0 ) {
		columns = 3;
	} else if ( fields.length === 1 ) {
		columns = 1;
	}
```